src/components/LeadsTable/leadsSlice.ts

```diff
@@ -0,0 +1,119 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+interface Lead {
+   _id: string;
+   name: string;
+   email: string;
+   consentsAccepted: boolean;
+   createdAt: string;
+   updatedAt: string;
+   __v: number;
+}
+
+interface LeadsState {
+   leads: Lead[];
+   isLoading: boolean;
+   error: string | null;
+   sortKey: 'name' | 'email' | 'date' | 'consentsAccepted';
```

Comment:
**issue:** such things as sortKey or sortOrder belong to UI layer, not data layer